lib/dao.js

```diff
@@ -1659,7 +1669,7 @@ DataAccessObject.find = function find(query, options, cb) {
             });
 
             // FIXME: apply "includes" and other transforms - see allCb below
-            memory.all(modelName, ctx.query, options, cb);
+            memory.all(modelName, query, options, cb);
```

Comment:
I think the idea here is that a hook can modify `ctx.query` and the changes will be picked up. Is that preserved by this change?

```js
MyModel.observe('access', function(ctx, next) {
  ctx.query = { /* new query */ };
  next();
});
```